server/api/attendance/attendance.controller.js

```diff
@@ -0,0 +1,674 @@
+
+// *******************************************************
+// Verify users small group attendance given submission Id, available only to mentors
+// TODO: verify that mentor is part of the smallgroup
+// router.post('/present/:user/small', auth.hasRole('mentor'), controller.verifySmallAttendance);
+exports.verifySmallAttendance = function(req,res){
+    var userId = req.params.id;
+    return ClassYear.getCurrent(function(err, classYear){
+        var classYearId = classYear._id;
+        return getPresent(user._id, isoDateToTime(new Date()), classYear._id, function(userAttendance){
+            if (!userAttendance.length){
+                return res.send(400, 'No attendance found');
+            }
+            for (var a = 0; a < userAttendance.length; a++){
+                if (userAttendance.smallgroup){
+                    userAttendance[a].verified = true;
+                    userAttendance[a].save();
+                }
+            }
+            return res.send(200);
+        });
+    });
+};
+// *******************************************************
+
+
+// *******************************************************
+// Get all attendance for a specific user (or current user) in the current classyear
+// router.get('/present/me', auth.isAuthenticated(), controller.getAttendanceMe);
+// router.get('/present/:user', auth.hasRole('mentor'), controller.getAttendance);
+
+// Get all attendance for a specific user (or current user) in the current class year
+var getAttendance = function(userId, classYearId, cb){
+    var callback = cb || function(){};
+    Attendance.find({user:userId, classYear:classYearId}, function (err, attendance) {
+        return callback(attendance);
+    });
+};
+exports.getAttendance = function(req, res) {
+    var userId = req.params.user;
+    return ClassYear.getCurrent(function(err, classYear){
+        var classYearId = classYear._id;
+        getAttendance(userId, classYearId, function(userAttendance){
+            return res.json(userAttendance);
+        });
+    });
+};
+exports.getAttendanceMe = function(req, res) {
+    req.params.user = req.user._id
+    exports.getAttendance(req,res);
+};
+// *******************************************************
+
+
+// *******************************************************
+// Get attendance for a specific user (or current user) on a date
+//
+// router.get('/present/:user/today', auth.hasRole('mentor'), controller.present);
+// router.get('/present/:user/:date', auth.hasRole('mentor'), controller.present);
+//
+// router.get('/present/me/today', auth.isAuthenticated(), controller.presentMe);
+// router.get('/present/me/:date', auth.isAuthenticated(), controller.presentMe);
+
+// Check if a userId is present today
+var getPresent = function(userId, date, classYearId, cb){
+    var callback = cb || function(){};
+    Attendance.find({user:userId, date: date, classYear:classYearId}, function (err, attendance) {
```

Comment:
No. A user can have multiple attendances on the same day (full group, and a bonus activity, ect)